tkp/db/alchemy/image.py

```diff
@@ -0,0 +1,192 @@
+import math
+from datetime import datetime
+from tkp.db.model import Frequencyband, Skyregion, Image, Dataset
+from tkp.utility.coordinates import eq_to_cart
+from sqlalchemy import text, func
+
+
+def get_band(session, dataset, freq_eff, freq_bw, freq_bw_max=.0):
+    """
+    Returns the frequency band for the given frequency parameters. Will create a new frequency band entry in the
+    database if no match is found. You can limit the bandwidth of the band association with the freq_bw_max.
+
+    args:
+        session (sqlalchemy.orm.session.Session): a SQLAlchemy session object
+        dataset (tkp.db.model.Dataset): the TraP dataset
+        freq_eff (float): The central frequency of image to get band for
+        freq_bw (float): The bandwidth of image to get band for
+        freq_bw_max (float): The maximum bandwith used for band association. Not used if 0.0 (default).
+
+    returns:
+        tkp.db.model.Frequencyband: a frequency band object
+    """
+    
+    if freq_bw_max == .0:
+        bw_half = freq_bw / 2
+        low = str(freq_eff - bw_half)
+        high = str(freq_eff + bw_half)
+        filter_ = " between freq_low and freq_high"
+    else:
+        bw_half = freq_bw_max / 2
+        low = str(freq_eff - bw_half)
+        high = str(freq_eff + bw_half)
+        filter_ = " between freq_central - %s and freq_central + %s" % (bw_half, bw_half)
+
+    band = session.query(Frequencyband). \
+        filter((Frequencyband.dataset == dataset) & (text(low + filter_) | text(high + filter_))).first()
```

Comment:
or we test on `max(low, freq_low) <= min(high, freq_high)`, that always returns the overlapping bands. what do you think?